src/modules/site-v2/base/views/strains.py

```diff
@@ -342,16 +346,20 @@ def order_confirmation(invoice_hash):
     abort(404)
 
   # Parse the individual items in the order into a list of dicts
-  order_obj["items"] = [ x for x in order_obj['items'].split("\n") ]
   items = []
-  for row in order_obj['items']:
+  for row in order_obj['items'].split("\n"):
     arr = row.split(', ')
     item_dict = {}
     for x in arr:
       k, v = x.split(': ')
       if k == 'price':
         v = float(v)
       item_dict[k] = v
+    species = item_dict.get('species')

```

Comment:
Aha, I like the idea here!  This is definitely a valid solution, and I don't want to fixate on smaller details when we have so much else to work on.

That said -- I wonder if it's possible to add the `Species` object to the `item_dict` directly? e.g. something like:
```
item_dict['species'] = Species.from_name(species) if species else None
```
Then the full `Species` object would be available in the template.

Might not be worth doing now, but I'd at least hold onto the idea as an improvement for later.  It might make sense to turn the cart items into their own `class` at some point, and if so, you could make this change then.

Reply:
I like the idea, I'll create a BC card for this and put it to later ToDos